### PR TITLE
feat: Add JWT to token response

### PR DESCRIPTION
**Pull request by loganripplinger**
# Description

## JIRA ticket

[HIVE-1644]

## Summary

This is the preliminary implementation for Authserver sending meaningful JWT's to the frontend. Authserver still needs to reach out to the Permissions service to get the scopes.

The token response still contains the access_token (as does the JWT).

After exploring through authlib, I couldn't find a more "proper" way of integrating the JWT. This actually seems to be likely the right way to do it.

- `/token` now responds with

```json
{
    "access_token": "q37J1LHfrPuPNxUrmLZ1VDzNLP41O62INEJbR9KfVk",
    "expires_in": 864000,
    "jwt": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJpc3MiOiJicmlnaHRoaXZlLWF1dGhzZXJ2ZXIiLCJhdWQiOiJicmlnaHRoaXZlLXBsYXRmb3JtLWFwaXMiLCJpYXQiOjE2MjE0NzY5MjcsImV4cCI6MTYyMzU1MDUyNywiYnJpZ2h0aGl2ZS1hY2Nlc3MtdG9rZW4iOiJxMzdKMUxIZnJQdVBOeFVybUxaMVZEek5MUDQxTzYySU5FSmJSOUtmVmsiLCJicmlnaHRoaXZlLW9yZy1yb2xlIjoiPz8_In0.YG8C4-TwgYvHkcyem1bjRSS5-kX-YQYrsATTdAW0qrEqbWH-8C-1tw6hBWe6MNOpPm8BTjV4kG44Z624Bo6mDfRYOpLJ13ztWXZFrP9PXpijVM6PxnVsC4d-7xH1KImR_ItkcCUcErkGTbKVwB2k1sqZbJWYl49t87AZ40bZgrac3waqGQ2_f3y7zy9YBN1xuy0h3fYeXBnu2gHT7nK7lREFFOm3l3lDNauN9ZZ7FF26gaGBDA9iLlVqsPjYlj8HAbNWCkNCdoI-3c0DK1aJjqJGi6vIAUc6ANPnxOvyqldz50R80n-xx2U18BQtjObl1fNG7a40B_z2gc5_SBaw37qX7NSnnhKSSL5pZ6PDCqfCeAoYjrjLiAzGh6Acq9ibcmUI8vr2VMjHM4Z8n0FltK0S4hgxTS2y82_QeyG3mecYqolYY7k1WVRzbprI9N0N2WsyJ0m0jg_d71CPB-5EDwoyih_Bj3nb4KqfKYZ-J7-AwK9prEiT_ifrbcDZG177C1sMZNUN2aX9dfW_2QZzfQK4L0v9sqUXRMfwJfIwNiBn_inDpHoryHZ6ozeauHK2_dntbfFtu-61P0wdYqJblK6kBp3bU13-yTTxDm-fG0zmWjl9RfGdLBfIy6uhK9bIVal3-CI4pUzFTP90vgSnNXxIYSYQo39_4CSGDgUVwD8", 
    "token_type": "Bearer"
}
```

- Adds `run-locally.sql` which allows you to set the DB to a state that allows Facet to go through the OAuth flow.

- Pins flask version to `<2`

- fixes a lock file generation issue.

# Checklists

### Basic

- [ ] Did you write tests for the code in this PR?
- [ ] Did you document your changes in the README and/or in docstrings (as needed)?

### Security

- [ ] Authorization has been implemented across these changes
- [ ] Injection has been prevented (parameterized queries, no eval or system calls)
- [ ] Any web UI is escaping output (to prevent XSS)
- [ ] Sensitive data has been identified and is being protected properly

### Frontend

- [ ] HTTP links should embed text that clearly explains what information your readers will find when they click on a hyperlink. [Read more about HTML accessibility](https://accessibility.umn.edu/core-skills/hyperlinks).

# Notes for the Reviewer

[HIVE-1644]: https://brighthiveio.atlassian.net/browse/HIVE-1644

### CODE REVIEW

File: authserver/api/health.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 from flask_restful import Resource, Api
 from injector import inject
 from authserver.config import AbstractConfiguration
-
+# from authserver.oauth2.rfc6749.authorization_server import BrighthiveJWT
```
Comment (loganripplinger) [inline authserver/api/health.py:11]:
```suggestion
```

File: authserver/api/oauth2.py [hunk 2]
```diff
@@ -94,7 +94,7 @@ def authorize():
     return authorization.create_authorization_response(grant_user=user)
 
 
-class ValidateOAuth2TokenResource(Resource):
+class ValidateOAuth2TokenResource(Resource): # TODO: update
```
Comment (loganripplinger) [inline authserver/api/oauth2.py:97]:
```suggestion
class ValidateOAuth2TokenResource(Resource):
```

File: authserver/db/models/models.py [hunk 3]
```diff
@@ -224,7 +224,7 @@ class OAuth2AuthorizationCode(db.Model, OAuth2AuthorizationCodeMixin):
     code_challenge_method = db.Column(db.VARCHAR())
 
 
-class OAuth2Token(db.Model, OAuth2TokenMixin):
+class OAuth2Token(db.Model, OAuth2TokenMixin): # TODO: this needs to be updated? OAuth2TokenMixin makes that more complicated.
```
Comment (loganripplinger) [inline authserver/db/models/models.py:227]:
```suggestion
class OAuth2Token(db.Model, OAuth2TokenMixin):
```

File: authserver/utilities/oauth2.py [hunk 4]
```diff
@@ -94,13 +94,13 @@ class ClientCredentialsGrant(grants.ClientCredentialsGrant):
 )
 require_oauth = ResourceProtector()
 
-
+# TODO: Verify: None of these need to be modified?
```
Comment (loganripplinger) [inline authserver/utilities/oauth2.py:97]:
```suggestion
```

File: authserver/utilities/oauth2.py [hunk 5]
```diff
@@ -94,13 +94,13 @@ class ClientCredentialsGrant(grants.ClientCredentialsGrant):
...
 def config_oauth(app):
     authorization.init_app(app)
     authorization.register_client_auth_method(
         'client_secret_json', authenticate_client_secret_json)
 
-    # supported grant types
+    # supported grant types # TODO: grant.create_token_response() is calling this
```
Comment (loganripplinger) [inline authserver/utilities/oauth2.py:103]:
```suggestion
    # supported grant types
```

File: authserver/app/app.py [hunk 6]
```diff
@@ -33,6 +33,7 @@ def teardown_appcontext(_):
         g.graph_db.close()
 
 
+
```
Comment (loganripplinger) [inline authserver/app/app.py:36]:
```suggestion
```

File: authserver/oauth2/rfc6749/authorization_server.py [hunk 7]
```diff
@@ -72,7 +108,13 @@ def create_token_response(self, request=None):
 
         try:
             grant.validate_token_request()
-            args = grant.create_token_response()
-            return self.handle_response(*args)
+            status, body, headers = grant.create_token_response()
+
+            bh_jwt = generate_jwt(body['access_token'])
```
Comment (loganripplinger) [inline authserver/oauth2/rfc6749/authorization_server.py:113]: @gregmundy this _seems_ to be okay based on all the code I went through in authlib.

File: authserver/config/config.py [hunk 8]
```diff
@@ -64,6 +63,19 @@ def __init__(self):
...
 
+        self.permission_service_url = os.getenv('PERMISSION_URL', 'http://localhost:2001')
+        signature_private_path = os.getenv('SIGNATURE_PRIVATE_PATH', None)
+
+        if signature_private_path is None:
+            print("Private key for JWT signature not defined.")
+        
+        try:
+            with open(signature_private_path, "rb") as key_file:
+                self.signature_key = key_file.read()
```
Comment (loganripplinger) [inline authserver/config/config.py:74]: This should be cached and not reading the file each time its referenced.

File: Pipfile [hunk 9]
```diff
@@ -10,13 +10,12 @@ docker = "*"
...
 pytest-cov = "*"
 expects = "*"
-sphinx = "*"
 doc8 = "*"
 pytest-flask = "*"
 pytest-mock = "*"
 
 [packages]
-flask = "*"
+flask = "<2"
```
Comment (johnosullivan) [inline Pipfile:18]: 👍

File: authserver/oauth2/rfc6749/authorization_server.py [hunk 10]
```diff
@@ -8,6 +8,50 @@
...
+
+def generate_jwt(access_token):
+    try:
+        json_claims = {"brighthive-access-token": access_token, "brighthive-org-role": "???"}
+        a_jwt = BrighthiveJWT().make_jwt(json_claims)
+    except Exception as e:
+        print('JWT generation exception:', e)
+        return 'none'
+
+    return a_jwt
```
Comment (johnosullivan) [inline authserver/oauth2/rfc6749/authorization_server.py:54]: What is with the "???"?

Comment (loganripplinger) [reply authserver/oauth2/rfc6749/authorization_server.py:54]: just a placeholder for the lambda service

**Pull request merged**